src/bootstrap-table.js

```diff
@@ -1558,6 +1560,47 @@
             }
 
             html.push('</tr>');
+
+            if (detailViewColPos > -1) {
+
+                var existingRowIndex;           //Current item in the existing table
+                var $existingRow;               //jquery obj of the existing table item row
+                var $existingRowDetailView;    //jquery obj of the existing table item row's detailview
+                var $detailRow;                 //the new detail row we'll append to html
+
+                //if the detailview was open previously, keep it open
+                existingRowIndex = $.inArray(that.getRowByUniqueId(item[that.options.uniqueId]), that.options.data);
```

Comment:
In many cases, do not set the `uniqueId` option, it will cause an bug: https://jsfiddle.net/wenyi/kgL1rocs/2/.